src/frontend/Deprecation_analysis.ml

```diff
@@ -137,6 +137,40 @@ let rec collect_deprecated_expr (acc : (Location_span.t * string) list)
                 ) ]
           | _ -> [] ) in
       acc @ w @ List.concat_map l ~f:(fun e -> collect_deprecated_expr [] e)
+  | PrefixOp (PNot, ({emeta= {type_= UReal; loc; _}; _} as e)) ->
+      let acc =
+        acc
+        @ [ ( loc
+            , "Using a real as a boolean value is deprecated and will be \
```

Comment:
I feel like it would be good to potentially note that exact comparison of a float to zero is also a bad idea, and they should consider `abs(x) < epsilon` or something else. I'm not sure of the exact wording, but it feels weird to me to deprecate this but recommend something with the same poor behavior

Reply:
Yes, something like that definitely makes sense. I'm not sure about exact wording either and it probably should trigger on all uses of `==` and `!=`, not just the implicit ones.
But I think comparison with _zero_ in particular is more justified, at least for DataOnly, because it might be used as a marker value indicating missing data.